ibm/resource_ibm_database.go

```diff
@@ -1721,63 +1755,77 @@ func resourceIBMDatabaseInstanceUpdate(d *schema.ResourceData, meta interface{})
 		remove := os.Difference(ns).List()
 		add := ns.Difference(os).List()
 
+		if len(remove) > 0 {
+			for _, entry := range remove {
+				newEntry := entry.(map[string]interface{})
+				deleteDatabaseUserOptions := &clouddatabasesv5.DeleteDatabaseUserOptions{
+					ID:       &instanceID,
+					UserType: core.StringPtr(newEntry["type"].(string)),
+					Username: core.StringPtr(newEntry["name"].(string)),
+				}
+
+				deleteDatabaseUserResponse, _, err := cloudDatabasesV5.DeleteDatabaseUser(deleteDatabaseUserOptions)
+				if err != nil {
+					return fmt.Errorf("[ERROR] Error deleting database user (%s) entry: %s", *deleteDatabaseUserOptions.Username, err)
+				}
+				taskID := *deleteDatabaseUserResponse.Task.ID
+				_, err = waitForDatabaseTaskComplete(taskID, d, meta, d.Timeout(schema.TimeoutUpdate))
+				if err != nil {
+					return fmt.Errorf(
+						"[ERROR] Error waiting for database (%s) user (%s) delete task to complete: %s", icdId, *deleteDatabaseUserOptions.Username, err)
+				}
+			}
+		}
+
 		if len(add) > 0 {
 			for _, entry := range add {
 				newEntry := entry.(map[string]interface{})
-				userEntry := icdv4.User{
-					UserName: newEntry["name"].(string),
-					Password: newEntry["password"].(string),
+				userEntry := &clouddatabasesv5.CreateDatabaseUserRequestUser{
+					Username: core.StringPtr(newEntry["name"].(string)),
+					Password: core.StringPtr(newEntry["password"].(string)),
+				}
+
+				createDatabaseUserOptions := &clouddatabasesv5.CreateDatabaseUserOptions{
+					ID:       &instanceID,
+					UserType: core.StringPtr(newEntry["type"].(string)),
+					User:     userEntry,
 				}
-				userReq := icdv4.UserReq{
-					User: userEntry,
+
+				createDatabaseUserResponse, _, err := cloudDatabasesV5.CreateDatabaseUser(createDatabaseUserOptions)
+				if err != nil {
```

Comment:
We are returning the error if we have any error due to createDatabse the line 1802 will never be executed if the user already exists and update the password